topics/appendix/tools/tools_gradle_intellij_plugin.md

```diff
@@ -0,0 +1,1210 @@
+
+**Type:** `String`
+
+**Default value:** `${project.version}`
+
+
+### changeNotes
+The change notes of the plugin – will be set for the `<change-notes>` tag.
+
+**Type:** `String`
+
+**Default value:** `null`
+
+
+### pluginId
+The ID of the plugin – will be set for the `<id>` tag.
+
+**Type:** `String`
+
+**Default value:** `null`
+
+
+## prepareSandbox Task
+Prepares sandbox directory with installed plugin and its dependencies.
+
+
+### pluginName
+The name of the plugin.
+
+**Type:** `String`
+
+**Default value:** `${intellij.pluginName}`
+
+
+### configDir
+The directory with the plugin configuration.
+
+**Type:** `String`
+
+**Default value:** `${intellij.pluginName}/config`
+
+
+### pluginJar
+The input plugin JAR file used to prepare the sandbox.
+
+**Type:** `File`
+
+**Default value:** `jar` task output
+
+
+### librariesToIgnore
+Libraries that will be ignored when preparing the sandbox. By default, excludes all libraries that are a part of the `setupDependenciesTask.idea` dependency.
+
+**Type:** `List<File>`
+
+**Default value:** `org.jetbrains.intellij.tasks.SetupDependenciesTask.idea.get().jarFiles`
+
+
+### pluginDependencies
+List of dependencies of the current plugin.
+
+**Type:** `List<PluginDependency>`
+
+**Default value:** `org.jetbrains.intellij.IntelliJPluginExtension.getPluginDependenciesList`
+
+
+## prepareTestingSandbox Task
+Prepares sandbox directory with installed plugin and its dependencies for testing purposes.
+
+See [`prepareSandbox` Task](#preparesandbox-task).
+
+
+## prepareUiTestingSandbox Task
+Prepares sandbox directory with installed plugin and its dependencies for UI testing purposes.
+
+See [`prepareSandbox` Task](#preparesandbox-task).
+
+
+## publishPlugin Task
+Publishes plugin to the remote Marketplace repository.
+
+The following attributes are a part of the Publishing DSL `publishPlugin { ... }` in which allows Gradle to upload a working plugin to the JetBrains Plugin Repository. Note that you need to upload the plugin to the repository at least once manually (to specify options like the license, repository URL etc.) before uploads through Gradle can be used.
+
+See the instruction on [how to generate authentication token](https://plugins.jetbrains.com/docs/marketplace/plugin-upload.html).
+
+See [Publishing Plugins with Gradle](deployment.md) tutorial for step-by-step instructions.
+
+
+### token
+Authentication token.
+
+**Type:** `String`
+
+**Default value:** `null`
+
+
+### channels
+List of channel names to upload plugin to.
+
+**Type:** `List<String>`
+
+**Default value:** `["default"]`
+
+
+### host
+URL host of a plugin repository.
+
+**Type:** `String`
+
+**Default value:** `https://plugins.jetbrains.com`
+
+
+### distributionFile
+Jar or Zip file of plugin to upload.
+
+**Type:** `File`
+
+**Default value:** output of the [`buildPlugin`](#buildplugin-task) task
+
+
+### toolboxEnterprise
+Specifies if the Toolbox Enterprise plugin repository service should be used.
```

Comment:
explain/link what this is

Reply:
ATM there's nothing to explain.

Reply:
then we should clarify this is not for general public use  now